src/recurrer.ts

```diff
@@ -1,111 +1,155 @@
-import _ from 'lodash'
+import * as _ from 'lodash'
+import { ADDITIONAL_UNIT, DAYS_IN_WEEK, HOUR_IN_DAY, MIN_IN_HOUR, MS_IN_SEC, NO_DIFF, SEC_IN_MIN } from './constants'
 
-// For producing recurring timestamps 
-export default class Recurrer {
-  async getDailyRecurringTimestamps(
-    startTimestamp: number,
-    numberRecurring: number,
-    hourOfDay: HourOfDay,
-  ): Promise<number[]> {
+// For producing recurring timestamps
+export class Recurrer {
+  /**
+   * Input starting timestamp in milliseconds along with desired hour of day.
+   * Output up to 24 daily recurring timestamps in milliseconds
+   * @param startTimestamp
+   * @param numberRecurring
+   * @param hourOfDay
+   * @returns daily recurring timestamps
+   */
+  getDailyRecurringTimestamps(startTimestamp: number, numberRecurring: number, hourOfDay: HourOfDay): number[] {
     const startDate = new Date(startTimestamp)
     const startHour = startDate.getUTCHours()
     const startYear = startDate.getUTCFullYear()
     const startMonth = startDate.getUTCMonth()
-    const startDay = (startHour < hourOfDay) ? startDate.getUTCDate() : startDate.getUTCDate() + 1
+    const startDay = startHour < hourOfDay ? startDate.getUTCDate() : startDate.getUTCDate() + ADDITIONAL_UNIT
     const firstEventTimestamp = Date.UTC(startYear, startMonth, startDay, hourOfDay)
-    const secondsInDay = 24 * 60 * 60
-    return _.times(numberRecurring, index => firstEventTimestamp + (index * secondsInDay))
+    const milliSecondsInDay = HOUR_IN_DAY * MIN_IN_HOUR * SEC_IN_MIN * MS_IN_SEC
+    return _.times(numberRecurring, (index) => {
+      return firstEventTimestamp + index * milliSecondsInDay
+    })
   }
 
-  async getHourlyRecurringTimestamps(
-    startTimestamp: number,
-    numberRecurring: number,
-  ): Promise<number[]> {
-    const secondsInHour = 60 * 60
+  /**
+   * Input starting timestamp in milliseconds.
+   * Output up to 24 hourly recurring timestamps in milliseconds
+   * @param startTimestamp
+   * @param numberRecurring
+   * @returns hourly recurring timestamps
+   */
+  getHourlyRecurringTimestamps(startTimestamp: number, numberRecurring: number): number[] {
+    const secondsInHour = MIN_IN_HOUR * SEC_IN_MIN * MS_IN_SEC
     const firstEventTimestamp = startTimestamp - (startTimestamp % secondsInHour) + secondsInHour
```

Comment:
To make sure I am reading this right, if the hour 12:30 is passed then the `firstEventTimestamp` will be 13:00?

Reply:
I understand trying to make this easier for "lazy" devs but this could also get confusing as it's not clear that the timestamp will be rounded to the next hour.  This could be solved with clear documentation but I also think throwing an error with a very clear message that an "hour" timestamp is required could be fine, too.  And that would remove any uncertainty on when your task is going to start.

Reply:
i'll def put more documentation that'll explain the rounding. i was going to start with having the user input a timestamp that was the first one of the list. but also didn't necessarily want the users to have to do the calculations (like for finding the 3rd day of the 4th week of the month for the current month). I wanted them to be able to just insert Date.now() and be able to get an array of timestamps that they wanted.